Back-To-School/Assets/BackToSchool/Scripts/Progression/SaveSystem.cs

```diff
@@ -1,44 +1,35 @@
-﻿using Assets.BackToSchool.Scripts.Constants;
+﻿using Assets.BackToSchool.Scripts.Player;
 using UnityEngine;
 
 
 namespace Assets.BackToSchool.Scripts.Progression
 {
-    internal class SaveSystem
+    public class SaveSystem
     {
-        private Player.Player _player;
-        private StatsManager _statsManager;
+        public bool IsSaveDataExists() => PlayerPrefs.GetInt(Constants.SaveParams.IsSaveDataExists.ToString()) == 1;
 
-        public SaveSystem(Player.Player player, StatsManager statsManager)
+        public void SavePlayerProgress(PlayerController player, LevelSystem levelSystem)
```

Comment:
TODO for future improvements - work with interfaces or even other data structures that should handle only data that needs to be saved/loaded